### DISCUSSION TITLE
Jetstream instances don't automatically reconnect?

### DISCUSSION THREAD

**Discussion by jkwatson-verta** [nats-io/nats.java#677] [Q&A]
We're implementing our first java jetstream publishers and consumers, and have discovered that our `JetStream` instances don't automatically reconnect when the server becomes temporarily unavailable. Note that this isn't the behavior of a NATS `Connection` ..it does reconnect automatically, so I was expecting the resulting `JetStream` to do the same. 

Is this expected behavior? 

I'm testing it by having a local NATS server running in docker, setting up a couple of java processes to talk to it, then rapidly killing the server and bringing it back up (~1s of downtime altogether).

**Comment by scottf** [comment]
Are you sure, this is a pretty basic feature? Might have to do with docker and networking. If you can demonstrate it outside of docker please let me know.

**Reply by jkwatson-verta** [reply]
Hmm. interesting. When I run NATS locally (not in docker) and restart the server, everything does reconnect. Also, when I simple issues a `docker restart` on the NATS container, it *also* reconnects properly. Not sure what's going on with the docker kill/restart and why that isn't working, but it does seem like the library behaves as it should. 

Thanks and sorry for the trouble!
